src/tequila/grouping/binary_rep.py

```diff
@@ -230,23 +232,54 @@ def anti_commutativity_matrix(self):
         gram = np.block([[np.zeros((n,n)), np.eye(n)], [np.eye(n), np.zeros((n,n))]])
         return matrix @ gram @ matrix.T % 2
 
-    def commuting_groups(self, method='rlf'):
+    def commuting_groups(self, method='rlf', condition="fc", overlap_aux=None):
         """
+        Notes
+        ----------
         Return the partitioning of the hamiltonian into commuting groups.
+
+        Parameters
+        ----------
+        method: str: The method used to group the commutative terms together. In the case of overlapping methods,
```

Comment:
would be good to list the possibie keys here: e.g. rlf (with explanation), other (with explanation) .....